File: macos/Scanner/Sources/Scanner/Rules/AESAUTH005.swift
public struct AESAUTH005: Rule {
    public let id = "AES-AUTH-005"
    public let severity: Severity = .critical

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.mcpServers.compactMap { server in
            guard !server.disabled else {
                return nil
            }
            let match = server.searchableValues.first { value in
                AiThreatPatterns.isConnectionStringWithCredentials(value)
            }
            guard let match else {
                return nil
            }
            return finding(
                app: server.appId,
                serverName: server.name,
                affectedPath: server.configPath,
                maskedValue: SecretPatterns.masked(match)
            )
        }
    }
}

File: windows/Scanner/Rules/AESAUTH005.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESAUTH005 : IRule
{
    public string Id => "AES-AUTH-005";
    public Severity Severity => Severity.Critical;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts)
    {
        var findings = new List<Finding>();
        foreach (var server in facts.McpServers.Where(server => !server.Disabled))
        {
            var match = server.SearchableValues()
                .FirstOrDefault(AiThreatPatterns.IsConnectionStringWithCredentials);
            if (match is null)
            {
                continue;
            }
            findings.Add(this.Finding(
                server.AppId,
                server.Name,
                affectedPath: server.ConfigPath,
                maskedValue: SecretPatterns.Masked(match)
            ));
        }
        return findings;
    }
}
